gene_finder.py

```diff
@@ -142,25 +194,34 @@ def coding_strand_to_AA(dna):
         dna: a DNA sequence represented as a string
         returns: a string containing the sequence of amino acids encoded by the
                  the input DNA fragment
-
         >>> coding_strand_to_AA("ATGCGA")
         'MR'
         >>> coding_strand_to_AA("ATGCCCGCTTT")
         'MPA'
     """
-    # TODO: implement this
-    pass
-
+    a="" #New List for results to be put into
+    for i in range(0,len(dna),3): #Tell the function how long to look for
+        codon = dna[i:i+3] #Tell the function where to look
+        amino_acid = aa_table[codon] #assign value to amino_acid
+        a = a + amino_acid #continue the string
+    return a #actually return the amino acid
 
 def gene_finder(dna):
     """ Returns the amino acid sequences that are likely coded by the specified dna
 
         dna: a DNA sequence
         returns: a list of all amino acid sequences coded by the sequence dna.
     """
-    # TODO: implement this
-    pass
+    a=[]  #creating an empty list
+    threshold = longest_ORF_noncoding(dna,1500) #Assign Value to threshold
+    Long_Orfs = len(longest_ORF(dna)) #assign value to Long_Orfs
+    if Long_Orfs>threshold): #Compare values
+        a.append(coding_strand_to_AA(dna)) #add to the list
+    dna = load_seq("./data/X73525.fa") #obtaining genes
+    print gene_finder(dna) #showing the list of Amino Acids
```

Comment:
This is a recursive function, but we don't really need recursion here.
`return dna` would be the better way of ending this function.
Then we call gene_finder(dna) in `__main__`